src/pages/ZNS/ZNS.tsx

```diff
@@ -0,0 +1,59 @@
+//- React Imports
+import { FC } from 'react';
+
+//- Hook Imports
+import { useSubdomainData } from '../../lib/hooks/useSubdomainData';
+import { useDomainData } from '../../lib/hooks/useDomainData';
+import { usePaymentTokenInfo } from '../../lib/hooks/usePaymentTokenInfo';
+import { usePaymentTokenForDomain } from '../../lib/hooks/usePaymentTokenForDomain';
+import { useDomainMetadata } from '../../lib/hooks/useDomainMetadata';
+import { useDomainMetrics } from '../../lib/hooks/useDomainMetrics';
+import { useViewNavigation } from '../../lib/hooks/useViewNavigation';
+
+//- Container Imports
+import Domains from '../Domains/Domains';
+import NFT from '../NFT/NFT';
+
+//- Utils Imports
+import { getDomainId } from '../../lib/util/domains/domains';
+
+type ZNSProps = {
+	route: string;
+};
+
+const ZNS: FC<ZNSProps> = ({ route }) => {
+	const domainId = getDomainId(route);
+	const { data: domain } = useDomainData(domainId);
+	const { data: subdomainData } = useSubdomainData(domainId);
+	const { data: paymentToken } = usePaymentTokenForDomain(domainId);
+	const { data: paymentTokenInfo } = usePaymentTokenInfo(paymentToken);
+	const { data: metrics } = useDomainMetrics(domainId);
+	const { data: domainMetadata } = useDomainMetadata(domain?.metadataUri);
```

Comment:
Haha I totally feel you on this one - it's getting quite hectic. Could be handy to write a wrapper hook which contains all of these - it will help clean up this file.

Reply:
Made a change to this - not sure if this is what you had in mind, its basic, but cleans up the file